### ISSUE TITLE
P5 won't play sound when iPhone in silent mode

### ISSUE THREAD

**Issue by epignosis567** [processing/p5.js#3907]
#### Nature of issue?

- x Found a bug
- [ ] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [x] Other (specify if possible)

#### Which platform were you using when you encountered this?

- x Mobile/Tablet (touch devices)
- [ ] Desktop/Laptop
- [ ] Others (specify if possible)

#### Details about the bug: 

- p5.js version: 0.9.0 July 1 2019
- Web browser and version: Mobile Safari
- Operating System: iOS 12.1.2
- Steps to reproduce this:

Sounds triggered in P5 won't play when iPhone is in silent mode. I'm new to coding but as far as I can tell Javascript will typically play sounds that are triggered by a user even if the phone is in silent mode. Most forum posts I've found are asking for ways to PREVENT sound from playing while in silent mode. Is this a bug. I've been searching for answers but the most I can tell is that it has something to do with Apple's AVAudioSession class, but by default it should play sound in silent mode. The relevant code I'm using is as follows:

function mousePressed() 
  {
  getAudioContext().resume() 
  soundVAR.play();
  }

**Comment by epignosis567** [comment]
Still wondering about this...

**Comment by lmccart** [comment]
I think that these two posts may address the behavior you're noticing. If this is your scenario, it looks like there's not much that can be done from the p5 end, because this is native iphone / safari/ web audio behavior. There are a few hackier workarounds mentioned in the posts:
https://stackoverflow.com/questions/21122418/ios-webaudio-only-works-on-headphones/46839941#46839941
https://stackoverflow.com/questions/40789136/ios-ringer-switch-mutes-web-audio

**Comment by epignosis567** [comment]
Hi. Please don’t close this thread. Those linked threads do not address the issue. I feel like no one actually read through my original post here...

**Comment by lmccart** [comment]
sorry @epignosis567 I thought it was resolved because we hadn't heard back from you. as far as I can tell, the posts describe the issue you're talking about and explain that it is an issue of the underlying web audio framework. can you help us out by explaining how your issue differs. I did read through the whole post you made but still can't quite understand. thank you.

**Comment by epignosis567** [comment]
OK thank you, I’m not at a computer right now but I will post back ASAP with a more detailed explanation.

**Comment by epignosis567** [comment]
So one example would be Tone js, works as described, sounds play one iOS devices even while the device ringer is set to mute. Also like I mentioned before most of the questions about js sound are from people trying to figure out how to STOP the sound from playing on muted iOS devices. So this has to be something written into P5 right? Why is it that P5 sounds don't play on muted iphones?

**Comment by lmccart** [comment]
As mentioned in the link before, the sound toggle switch does mute WebAudio, but it does not mute HTML5 tags. So it is possible that the tone.js example you're looking at is making use of the HTML5 audio rather than tone.js. As far as I can tell this is normal behavior so I don't think it's anything we can fix with p5.js. Do others have suggestions?

**Comment by limzykenneth** [comment]
Here a few links to look at:

https://github.com/Tonejs/Tone.js/issues/164
For tone.js the audio source is `<audio>` which will play even if phone is on silent.

https://mdn.github.io/webaudio-examples/audio-analyser/
MDN example of WebAudio API which doesn't use `<audio>` as its audio source and thus will not play when phone is on silent.

https://html5tutorial.info/html5-audio.php
Example of basic `<audio>` tag, which will play even if phone is on silent.

Basically unless the audio source is an `<audio>` tag, the sound will not play when the phone is on silent. To make p5.Sound play audio on silent would be to make the audio source a `<audio>` tag but I don't think all of the audio sources can be converted, I'm not sure.

**Comment by lmccart** [comment]
thanks @limzykenneth. closing this since this isn't an issue we can fix from within p5.js
